src/databricks/labs/ucx/hive_metastore/table_move.py

```diff
@@ -0,0 +1,292 @@
+        except NotFound as err:
+            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(err) or "[DELTA_TABLE_NOT_FOUND]" in str(err):
+                logger.error(f"Could not find table {from_table_name}. Table not found.")
+            else:
+                logger.error(f"Failed to move table {from_table_name}: {err!s}", exc_info=True)
+        return False
+
+    def _alias_table(
+        self,
+        from_catalog: str,
+        from_schema: str,
+        from_table: str,
+        to_catalog: str,
+        to_schema: str,
+    ) -> bool:
+        from_table_name = f"{from_catalog}.{from_schema}.{from_table}"
+        to_table_name = f"{to_catalog}.{to_schema}.{from_table}"
+        try:
+            self._create_alias_view(from_table_name, to_table_name)
+            self._reapply_grants(from_table_name, to_table_name, target_view=True)
+            return True
+        except NotFound as err:
+            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(err) or "[DELTA_TABLE_NOT_FOUND]" in str(err):
+                logger.error(f"Could not find table {from_table_name}. Table not found.")
+            else:
+                logger.error(f"Failed to alias table {from_table_name}: {err!s}", exc_info=True)
+            return False
+
+    def _reapply_grants(self, from_table_name, to_table_name, *, target_view: bool = False):
+        try:
+            grants = self._ws.grants.get(SecurableType.TABLE, from_table_name)
+        except NotFound:
+            logger.warning(f"removed on the backend {from_table_name}")
+            return
+        if not grants.privilege_assignments:
+            return
+        logger.info(f"Applying grants on table {to_table_name}")
+        grants_changes = []
+        for permission in grants.privilege_assignments:
+            if not permission.privileges:
+                continue
+            if not target_view:
+                grants_changes.append(PermissionsChange(list(permission.privileges), permission.principal))
+                continue
+            privileges = set()
+            for privilege in permission.privileges:
+                if privilege != Privilege.MODIFY:
+                    privileges.add(privilege)
+            if privileges:
+                grants_changes.append(PermissionsChange(list(privileges), permission.principal))
+
+        self._ws.grants.update(SecurableType.TABLE, to_table_name, changes=grants_changes)
+
+    def _recreate_table(self, from_table_name, to_table_name, *, is_managed: bool, del_table: bool):
+        drop_table = f"DROP TABLE {escape_sql_identifier(from_table_name)}"
+        if is_managed:
+            create_table_sql = (
+                f"CREATE TABLE IF NOT EXISTS {escape_sql_identifier(to_table_name)} "
+                f"DEEP CLONE {escape_sql_identifier(from_table_name)}"
+            )
+            logger.info(f"Creating table {to_table_name}")
+            self._backend.execute(create_table_sql)
+            if del_table:
```

Comment:
Let's be extra careful and compare the number of rows in the new and old table before we're doing destructive actions